docs/appendix-types.md

```diff
@@ -0,0 +1,91 @@
+# Appendix: Types of `svd-viewer` 
```

Comment:
Generally I think it would be better extracting these into a typescript file containing the interfaces. Then reference that from the docs and include it in the code. Otherwise you need to keep 2 sources in sync.